src/test/java/htsjdk/samtools/LongReadsTest.java

```diff
@@ -0,0 +1,227 @@
+        final IOPath tmpBAMPath = createBAMCopyWithIndex(PAC_BIO_READS);
+
+        // get all unmapped reads from the original BAM and the copy and compare the results
+        try (final ReadsDecoder originalDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(ReadsBundle.resolveIndex(PAC_BIO_READS));
+             final ReadsDecoder copyDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(ReadsBundle.resolveIndex(tmpBAMPath));
+            final CloseableIterator<SAMRecord> originalIt = originalDecoder.queryUnmapped();
+            final CloseableIterator<SAMRecord> copyIt = copyDecoder.queryUnmapped()) {
+            while (originalIt.hasNext()) {
+                final SAMRecord originalRecord = originalIt.next();
+                Assert.assertTrue(copyIt.hasNext());
+                final SAMRecord copyRecord = copyIt.next();
+
+                Assert.assertEquals(copyRecord, originalRecord);
+            }
+            Assert.assertFalse(copyIt.hasNext());
+        }
+    }
+
+    @Test
+    final public void testSortOnWrite() throws IOException {
+        final List<SAMRecord> samRecords = new ArrayList<>();
+        SAMFileHeader samFileHeader;
+
+        final ReadsDecoderOptions readsDecoderOptions = new ReadsDecoderOptions()
+                .setValidationStringency(ValidationStringency.STRICT)
+                .setDecodeEagerly(true);
+
+        // get the original header, and the original reads in memory
+        try (final ReadsDecoder originalDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(
+                             ReadsBundle.resolveIndex(PAC_BIO_READS),
+                             readsDecoderOptions)) {
+            samFileHeader = originalDecoder.getHeader();
+            for (final SAMRecord r : originalDecoder) {
+                samRecords.add(r);
+            }
+        }
+
+        // reverse the order of the records, and write them back out with presorted=false
+        final List<SAMRecord> recordsInReverseOrder = new ArrayList(samRecords);
+        Collections.reverse(recordsInReverseOrder);
+        Assert.assertNotEquals(samRecords, recordsInReverseOrder);
+        final IOPath sortedOutputPath = IOPathUtils.createTempPath("testSortOnWrite", ".bam");
+
+        try (final ReadsEncoder bamEncoder = HtsDefaultRegistry.getReadsResolver().getReadsEncoder(
+                new ReadsBundle(sortedOutputPath),
+                new ReadsEncoderOptions().setPreSorted(false))) {
+            bamEncoder.setHeader(samFileHeader);
+            for (final SAMRecord samRecord : recordsInReverseOrder) {
+                bamEncoder.write(samRecord);
+            }
+        }
+
+        // now read back in both the original and the newly sorted reads that we wrote out, and compare the order
+        try (final ReadsDecoder originalDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(
```

Comment:
I wonder if we should add methods to reduce the boiler plate.  I wish we had default values in java...

Reply:
Yeah we should come up with something to make this easier.